### ISSUE TITLE
Service metadata changes are not propagated to Consul

### ISSUE THREAD

**Issue by exFalso** [hashicorp/nomad#7312]
If filing a bug please include the following:

### Nomad version
```
$ nomad version
Nomad v0.10.1
```

### Operating system and Environment details

```
$ uname -a
Linux genesis 4.19.78 #1-NixOS SMP Mon Oct 7 16:57:29 UTC 2019 x86_64 GNU/Linux
```

### Issue
Purely metadata changes to a job's `service` stanza are not propagated to Consul as the job is applied. If other things like the tags are changed, then the metadata changes are propagated correctly (which is our current workaround now).

### Reproduction steps
* Create a job with a `service` stanza with some metadata in it.
* Apply job
* Change *only* the metadata
* Apply job
* Observe Consul metadata is still the old one

**Comment by exFalso** [comment]
I just found this is also true for the service name, changes to it are not propagated

**Comment by notnoop** [comment]
This looks like a bug in our in-place update handling indeed.  Thanks for reporting this issue.

**Comment by urjitbhatia** [comment]
@notnoop any idea when we can expect a patch?

**Comment by urjitbhatia** [comment]
@exFalso or anyone else still seeing this:
Updating a dummy key in `job -> meta` seems to trigger a refresh of service meta sent to consul 🤷 
```
  meta {
     DUMMY_DEPLOY_ID = 4
  }
```

**Comment by notnoop** [comment]
@urjitbhatia Sorry for the slow response.  This bug was fixed in 0.10.4 .

The bug was that the check whether update is required didn't take Meta into account: This has been fixed in https://github.com/hashicorp/nomad/commit/3aec97753cc9b0d3b492a11f6a230191ed6d641b#diff-8612d5e61f84976db572b3f38eb380c0R103 .
